### PR TITLE
Matrix Cleanup Step 2

### PR DESCRIPTION
**Pull request by siefkenj**
This pull request continues the matrix cleanup work of https://github.com/sympy/sympy/pull/11954 by separating basic matrix arithmetic from `MatrixBase` into `MatrixArithmetic` which `MatrixBase` inherits from.

### CODE REVIEW

**Comment by siefkenj** [comment]
@moorepants would you mind looking at this?

**Comment by moorepants** [comment]
I won't have any time until next weekend probably.

**Comment by jksuom** [comment]
@siefkenj Can you add some comments on how this is intended to work. I find it hard to see from the scattered changes.

**Comment by siefkenj** [comment]
@jksuom I have a long description in the accepted PR https://github.com/sympy/sympy/pull/11954 This is a continuation of that work.

**Comment by siefkenj** [comment]
@moorepants @smichr @jksuom would one of you be able to review this?

File: sympy/matrices/dense.py [hunk 1]
```diff
@@ -53,6 +45,9 @@ def __eq__(self, other):
         except AttributeError:
             return False
 
+    def __ne__(self, other):
```
Comment (jksuom) [inline sympy/matrices/dense.py]: Maybe this is not necessary (https://docs.python.org/3/reference/datamodel.html#object.__ne__):

    By default, __ne__() delegates to __eq__() and inverts the result unless it is NotImplemented.

Comment (siefkenj) [inline sympy/matrices/dense.py]: If I recall correctly, I had to add that to get it working in python2.  Python3 properly negates `__eq__`.

Comment (jksuom) [inline sympy/matrices/dense.py]: Ok. I didn't check Py2.

**Comment by jksuom** [comment]
What is the role of MatrixBase now?

**Comment by siefkenj** [comment]
The answer is twofold.  `MatrixBase` implements the required methods from `MatrixRequired`.  Secondly, `MatrixBase` is the class that inherits from everything.  If you want to subclass a matrix and have every method available, you inherit from `MatrixBase`.  But, for example, if you wanted to make an object that behaved like a matrix but where the idea of eigenvectors doesn't make sense, you would inherit just from `MatrixArithmetic` and friends instead of `MatrixBase`.  

Also, not everything has been moved out of matrix base yet, and I suspect there will be some miscellaneous methods that don't fit any other category.  I was planning on leaving those in `MatrixBase` though I suppose one could create a `MatrixMisc`.

**Comment by siefkenj** [comment]
Ping.  Could I get a review, please?

**Comment by asmeurer** [comment]
This looks good to me. We should perhaps add some documentation for the different base matrix classes and what they should be used for (to Sphinx).

**Pull request merged**